cmd/vm-informant/main.go

```diff
@@ -97,54 +102,103 @@ func main() {
 	}
 }
 
+const minWaitDuration = 5 * time.Second
+
 // runRestartOnFailure repeatedly calls this binary with the same flags, but with 'auto-restart'
 // removed.
 //
 // We execute ourselves as a subprocess so that it's possible to appropriately cleanup after
 // termination by various signals (or an unhandled panic!). This is worthwhile because we *really*
 // don't want to leave the cgroup frozen while waiting to restart.
-func runRestartOnFailure(args []string, cleanupHooks []func()) {
+func runRestartOnFailure(ctx context.Context, args []string, cleanupHooks []func()) {
 	selfPath := os.Args[0]
-
-	minWaitDuration := time.Second * 5
+	timer := time.NewTimer(0)
+	defer timer.Stop()
 
 	for {
 		startTime := time.Now()
-
-		cmd := exec.CommandContext(context.TODO(), selfPath, args...)
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
-		err := cmd.Run()
-
+		sig := make(chan struct{})
```

Comment:
maybe not relevant given the comment about raciness, but in future these should have more descriptive names ­— instead of answering "what is this" (a signalling channel), I find it more useful to answer "what does it mean" (letting us know when the process is done!). Here, I'd personally call it `cmdDone` or something.

(although afaict it isn't _really_ indicating anything concrete, other than "the goroutine that was going to send it a SIGTERM exited" - even though it _seems_ to be used approximately as "command finished"?)